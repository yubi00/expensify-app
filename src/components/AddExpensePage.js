import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { addExpense } from '../actions/expenses'

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense)
        this.props.history.push('/')
    }

    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                    buttonLabel='Add Expense'
                    onSubmit={this.onSubmit}/>
            </div>
        )
    }
}

const mapDispatchToProp = (dispatch) =>({
        addExpense: (expense) => dispatch(addExpense(expense))
})

export default connect(undefined, mapDispatchToProp)(AddExpensePage)