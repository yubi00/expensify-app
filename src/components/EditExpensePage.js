import React from 'react'
import ExpenseForm from '../components/ExpenseForm'
import { connect } from 'react-redux'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import RemoveModal from '../components/RemoveModal'

export class EditExpensePage extends React.Component {
    state = {
        modalOpen: false
    }

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/dashboard')
    }

    handleModal = () => {
        this.setState(() => ({
            modalOpen: true
        }))
    }

    handleCloseModal = () => {
        this.setState(() => ({
            modalOpen: false
        }))
        this.props.history.push('/dashboard')
    }
   
    onRemove = () => {
        this.props.startRemoveExpense({id: this.props.expense.id })
        this.handleCloseModal()
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                    buttonLabel='Save'
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                    />
                    <button className="button button--remove"  onClick={this.handleModal}>Remove</button>
                </div>
                <RemoveModal
                    modalOpen={this.state.modalOpen}
                    handleCloseModal={this.handleCloseModal}
                    onRemove={this.onRemove}
                />
                
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)