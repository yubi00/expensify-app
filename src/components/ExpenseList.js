import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import getVisibleExpenses from '../selectors/expenses'


export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
            <div className="list-body">
                    {props.expenses.length !== 0 ? props.expenses.map((expense) =>
                    <ExpenseListItem {...expense} key={expense.id} />) 
                : <div className="list-item list-item--message">
                    <span>No expenses</span>
                </div> 
                    }
            </div>
    </div>
)
//function that map store state to the props of the components
const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList)
