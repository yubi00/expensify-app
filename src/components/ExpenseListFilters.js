import React from 'react'
import { connect } from 'react-redux'
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (calenderFocused) => {
        this.setState(() => ({calenderFocused}))
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    onSortChange = (e) => {
        e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount()
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            className="text-input" 
                            type="text" 
                            placeholder="Search expenses"
                            value={this.props.filters.text} 
                            onChange={this.onTextChange}
                         />
                    </div>
                    <div className="input-group__item">
                        <select className="select" value={this.props.filters.sortBy} onChange={this.onSortChange}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                        startDate={this.props.filters.startDate}
                        startDateId={'startDate'}
                        endDate={this.props.filters.endDate}
                        endDateId={'endDate'}
                        onDatesChange={this.onDatesChange}
                        focusedInput={this.state.calenderFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                         />
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () =>  dispatch(sortByAmount()) 
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)