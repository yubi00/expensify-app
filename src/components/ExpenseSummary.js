import React from 'react'
import { connect } from 'react-redux'
import getVisibleExpenses from '../selectors/expenses'
import getTotalExpensess from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    return (
            <div>
                {expensesCount !== 0 && <h3>Viewing {expensesCount} {expenseWord} totalling {numeral(expensesTotal/100).format('$0,0.00')}</h3>}
            </div>
        )
}
const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters)
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getTotalExpensess(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)

