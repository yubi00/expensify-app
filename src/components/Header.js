import React from 'react'
import {  NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to="/" activeClassName="is-active" exact>dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
    </header>
)

export default Header