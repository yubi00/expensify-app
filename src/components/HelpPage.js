import React from 'react'

const HelpPage = () => (
    <div>
        <h1>This is from my Help page</h1>
    </div>
)

export default HelpPage