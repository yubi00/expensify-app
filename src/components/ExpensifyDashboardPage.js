import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpenseSummary'

const ExpensifyDashboardPage = () => (
    <div>
        <ExpenseSummary/>
        <ExpenseListFilters/>
        <ExpenseList/>
     </div>
)

export default ExpensifyDashboardPage