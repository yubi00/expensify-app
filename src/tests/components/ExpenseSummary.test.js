import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('should render ExpenseSummary page correctly for single expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={250}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseSummary correctly for multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={5} expensesTotal={1050}/>)
    expect(wrapper).toMatchSnapshot()
})