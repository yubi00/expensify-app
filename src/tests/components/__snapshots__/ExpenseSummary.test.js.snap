// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly for multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    5
     
    expenses
     totalling 
    $10.50
  </h3>
</div>
`;

exports[`should render ExpenseSummary page correctly for single expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $2.50
  </h3>
</div>
`;
